its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/SonarLintTest.java

```diff
@@ -59,50 +67,64 @@ class SonarLintTest {
-    StandaloneGlobalConfiguration sonarLintConfig = StandaloneGlobalConfiguration.builder()
-      .addPlugin(TestsUtils.PLUGIN_LOCATION.getFile().toPath())
-      .setSonarLintUserHome(TEMP)
-      .addEnabledLanguage(Language.PYTHON)
-      .setLogOutput((formattedMessage, level) -> {
-        /* Don't pollute logs */ })
-      .setModulesProvider(Collections::emptyList)
+  static void prepare() {
+    var sonarLintConfig = AnalysisEngineConfiguration.builder()
+      .setWorkDir(TEMP)
       .build();
-    sonarlintEngine = new StandaloneSonarLintEngineImpl(sonarLintConfig);
+
+    var logOutput = new LogOutput() {
+      @Override
+      public void log(String formattedMessage, Level level, @Nullable String stacktrace) {
+        /* Don't pollute logs */
+      }
+    };
+    SonarLintLogger.setTarget(logOutput);
+    var pluginJarLocation = Set.of(TestsUtils.PLUGIN_LOCATION.getFile().toPath());
+    var enabledLanguages = Set.of(SonarLanguage.PYTHON);
+    var pluginConfiguration = new PluginsLoader.Configuration(pluginJarLocation, enabledLanguages, false, Optional.empty());
+    var pluginLoader = new PluginsLoader().load(pluginConfiguration, Set.of());
+
+    sonarlintEngine = new AnalysisEngine(sonarLintConfig, pluginLoader.getLoadedPlugins(), logOutput);
     baseDir = TEMP.toFile();
   }
 
   @AfterAll
   static void stop() {
+    SonarLintLogger.setTarget(null);
     sonarlintEngine.stop();
   }
 
   @Test
-  void should_raise_issues() throws IOException {
+  void should_raise_issues() throws IOException, InterruptedException, ExecutionException {
     ClientInputFile inputFile = prepareInputFile("foo.py",
       "def fooBar():\n"
         + "  `1` \n"
         + "  `1` #NOSONAR\n",
       false);
 
     List<Issue> issues = new ArrayList<>();
-    StandaloneAnalysisConfiguration configuration =
-      StandaloneAnalysisConfiguration.builder()
-        .setBaseDir(baseDir.toPath())
-        .addInputFile(inputFile)
-        .setModuleKey("myModule")
-        .build();
-
-    Map<ClientLogOutput.Level, List<String>> logsByLevel = new HashMap<>();
-    ClientLogOutput logOutput = (s, level) -> {
-      List<String> logs = logsByLevel.getOrDefault(level, new ArrayList<>());
-      logs.add(s);
-      logsByLevel.putIfAbsent(level, logs);
+    var configuration = AnalysisConfiguration.builder()
+      .setBaseDir(baseDir.toPath())
+      .addInputFile(inputFile)
+      .addActiveRules(
+        new ActiveRule("python:S1542", SonarLanguage.PYTHON.name()),
+        new ActiveRule("python:BackticksUsage", SonarLanguage.PYTHON.name()))
+      .build();
+
+    Map<Level, List<String>> logsByLevel = new HashMap<>();
+    LogOutput logOutput = new LogOutput() {
+      @Override
+      public void log(String formattedMessage, Level level, @Nullable String stacktrace) {
+        List<String> logs = logsByLevel.getOrDefault(level, new ArrayList<>());
+        logs.add(formattedMessage);
+        logsByLevel.putIfAbsent(level, logs);
```

Comment:
I find the version above a bit more concise
```suggestion
        logsByLevel.computeIfAbsent(level, k -> new ArrayList<>()).add(formattedMessage);
```